pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -472,9 +472,16 @@ func (e *EndpointController) syncService(key string) error {
 
 	createEndpoints := len(currentEndpoints.ResourceVersion) == 0
 
+	// Compare the sorted subsets and labels
+	// Remove the HeadlessService label from the endpoints if it exists, as this won't be set on the service itself
+	// and will cause a false negative in this diff check. But first check if it has that label to avoid expensive copies.
+	compareLabels := currentEndpoints.Labels
+	if _, ok := currentEndpoints.Labels[v1.IsHeadlessService]; ok {
+		compareLabels = utillabels.CloneAndRemoveLabel(currentEndpoints.Labels, v1.IsHeadlessService)
+	}
 	if !createEndpoints &&
-		apiequality.Semantic.DeepEqual(currentEndpoints.Subsets, subsets) &&
-		apiequality.Semantic.DeepEqual(currentEndpoints.Labels, service.Labels) {
+		apiequality.Semantic.DeepEqual(endpoints.RepackSubsets(currentEndpoints.Subsets), subsets) &&
```

Comment:
We no longer want to repack here.

Reply:
Updated, and changed the commit message to reflect that we're just removing the Headless label if necessary now